psst-gui/src/ui/public_user.rs

```diff
@@ -0,0 +1,368 @@
+    })
+    .with_text_size(theme::TEXT_SIZE_SMALL)
+    .with_text_color(theme::PLACEHOLDER_COLOR);
+
+    let follow_unfollow_button = Button::dynamic(|info: &Arc<PublicUserInformation>, _| {
+        if info.is_following.unwrap_or(false) {
+            "Unfollow".to_string()
+        } else {
+            "Follow".to_string()
+        }
+    })
+    .disabled_if(|info: &Arc<PublicUserInformation>, _| {
+        !info.allow_follows || info.is_current_user.unwrap_or(false)
+    })
+    .on_click(|ctx, info: &mut Arc<PublicUserInformation>, _| {
+        // Extract username from URI (format: spotify:user:USERNAME)
+        let username: Arc<str> = Arc::from(info.uri.split(':').nth(2).unwrap_or(&info.uri));
+        if info.is_following.unwrap_or(false) {
+            ctx.submit_command(UNFOLLOW_USER.with(username));
+        } else {
+            ctx.submit_command(FOLLOW_USER.with(username));
+        }
+    });
+
+    let user_info = Flex::column()
+        .cross_axis_alignment(CrossAxisAlignment::Start)
+        .with_child(user_name)
+        .with_default_spacer()
+        .with_child(follower_count)
+        .with_spacer(theme::grid(0.5))
+        .with_child(following_count)
+        .with_default_spacer()
+        .with_child(follow_unfollow_button)
+        .padding(theme::grid(1.0))
+        .lens(Ctx::data().then(Cached::data));
+
+    Flex::row()
+        .with_spacer(theme::grid(4.2))
+        .with_child(user_cover)
+        .with_default_spacer()
+        .with_child(user_info)
+}
+
+fn user_playlists_widget() -> impl Widget<WithCtx<MixedView>> {
+    Either::new(
+        |playlists: &WithCtx<MixedView>, &_| playlists.data.playlists.is_empty(),
+        Empty,
+        Flex::column()
+            .with_child(
+                Label::new("Public Playlists")
+                    .with_text_size(theme::grid(2.5))
+                    .align_left()
+                    .padding((theme::grid(1.5), theme::grid(0.5))),
+            )
+            .with_child(
+                Scroll::new(List::new(|| playlist::playlist_widget(true)).horizontal())
+                    .horizontal()
+                    .align_left()
+                    .lens(Ctx::map(MixedView::playlists)),
+            ),
+    )
+}
+
+fn user_artists_widget() -> impl Widget<WithCtx<MixedView>> {
+    Either::new(
+        |artists: &WithCtx<MixedView>, &_| artists.data.artists.is_empty(),
+        Empty,
+        Flex::column()
+            .with_child(
+                Label::new("Recently Played Artists")
+                    .with_text_size(theme::grid(2.5))
+                    .align_left()
+                    .padding((theme::grid(1.5), theme::grid(0.5))),
+            )
+            .with_child(
+                Scroll::new(List::new(|| artist::artist_widget(true)).horizontal())
+                    .horizontal()
+                    .align_left()
+                    .lens(Ctx::data().then(MixedView::artists)),
+            ),
+    )
+}
```

Comment:
These can be a widget builder function to make these widgets with diff content and titles since they use all the same padding, margins etc.